### PR TITLE
feat: adds OSS Orama support

### PR DESCRIPTION
**Pull request by micheleriva**
Adds OSS Orama Support as option of Client Instance

### CODE REVIEW

**Comment by raiindev** [comment]
LGTM!

**Review by raiindev — APPROVED** [review]

File: packages/ui-stencil/package.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
   "types": "dist/types/index.d.ts",
   "dependencies": {
     "@orama/highlight": "^0.1.6",
-    "@orama/orama": "^2.0.23",
+    "@orama/orama": "3.0.0",
+    "@orama/switch": "3.0.0",
```
Comment (faustoq) [inline packages/ui-stencil/package.json:27]:
```suggestion
    "@orama/orama": "^3.0.0",
    "@orama/switch": "^3.0.0",
```

File: packages/ui-stencil/package.json [hunk 2]
```diff
@@ -23,7 +23,8 @@
...
     "@oramacloud/client": "1.3.17",
```
Comment (faustoq) [inline packages/ui-stencil/package.json:28]:
```suggestion
    "@oramacloud/client": "^1.3.17",
```

File: apps/storybook/stories/config.ts [hunk 3]
```diff
@@ -28,11 +28,6 @@ const demoIndexes: DemoIndexConfig = {
       description: 'content',
       section: 'category',
     },
-    highlight: {
```
Comment (rjborba) [inline apps/storybook/stories/config.ts:31]: Forgot to remove it. Thanks! You could also use highlightTitle or highlightDescription

**Review by rjborba — APPROVED** [review]

**Comment by raiindev** [comment]
With commit [c638a16](https://github.com/oramasearch/orama-ui-components/pull/39/commits/c638a1662c72e3678e67c3f5e27200470d934d3d), I've added some workarounds to make OSS Chat work.
The first one is related to `interaction.sources` being an object when the question in the chat returns no results.
Since it changes structure, I just added a ternary operator.

The second one is related to the `ask` method arguments; I've noticed that the OSS instance does not support some of the passed arguments, so I added a condition to pass those only in Cloud mode.

File: packages/ui-stencil/src/components.d.ts [hunk 4]
```diff
@@ -58,7 +58,7 @@ export namespace Components {
     }
     interface OramaChatBox {
         "autoFocus": boolean;
-        "clientInstance"?: OramaClient;
+        "clientInstance"?: OramaClient | Orama<unknown>;
```
Comment (rjborba) [inline packages/ui-stencil/src/components.d.ts:61]: Shouldn't we be using OramaSwtich?

File: packages/ui-stencil/src/components.d.ts [hunk 5]
```diff
@@ -133,7 +133,7 @@ export namespace Components {
     }
     interface OramaSearchBox {
         "chatPlaceholder"?: string;
-        "clientInstance"?: OramaClient;
+        "clientInstance"?: OramaClient | AnyOrama;
```
Comment (rjborba) [inline packages/ui-stencil/src/components.d.ts:136]: Why we're usingAnyOrama here and Oraam<unknon> on OramaChatBox?

File: packages/ui-stencil/src/components.d.ts [hunk 6]
```diff
@@ -517,7 +517,7 @@ declare namespace LocalJSX {
     }
     interface OramaChatBox {
         "autoFocus"?: boolean;
-        "clientInstance"?: OramaClient;
+        "clientInstance"?: OramaClient | Orama<unknown>;
```
Comment (rjborba) [inline packages/ui-stencil/src/components.d.ts:520]: Same as previous comment

File: packages/ui-stencil/src/components/orama-chat-box/orama-chat-box.tsx [hunk 7]
```diff
@@ -14,7 +15,7 @@ import '@phosphor-icons/webcomponents/dist/icons/PhArrowClockwise.mjs'
 export class ChatBox {
   @Element() el: HTMLElement
   @Prop() index?: CloudIndexConfig
-  @Prop() clientInstance?: OramaClient
+  @Prop() clientInstance?: OramaClient | Orama<unknown>
```
Comment (rjborba) [inline packages/ui-stencil/src/components/orama-chat-box/orama-chat-box.tsx:18]: Shouldn't we be using OramaSwitch?

File: packages/ui-stencil/src/components.d.ts (hunk 4 above)
Comment (raiindev) [reply packages/ui-stencil/src/components.d.ts:61]: The `clientInstance` coming as a prop should be an `OramaClient | AnyOrama.`
The Switch is instantiated inside the ChatService / SearchService.
Ex. [ChatService](https://github.com/oramasearch/orama-ui-components/pull/39/files#diff-618f2dd4aaac9973128dd2a0732ba3da71bc1586a01f5c9d8851714443c5957cR13)
[SearchService](https://github.com/oramasearch/orama-ui-components/pull/39/files#diff-6f968c09e991bb7dc36b60a61983e9fa98f9b24facf7b8a3b617bd5223c30062R24)

**Pull request merged**